crates/wasi-common/src/virtfs.rs

```diff
@@ -255,7 +255,14 @@ impl Handle for InMemoryFile {
     fn read_vectored(&self, iovs: &mut [io::IoSliceMut]) -> Result<usize> {
         trace!("read_vectored(iovs={:?})", iovs);
         trace!("     | *read_start={:?}", self.cursor.get());
-        self.data.borrow_mut().preadv(iovs, self.cursor.get())
+        let read = self.data.borrow_mut().preadv(iovs, self.cursor.get())?;
+        let update = self
+            .cursor
+            .get()
+            .checked_add(read as u64)
+            .ok_or(Errno::Fbig)?;
```

Comment:
I think this should either be `EFAULT` or `ENOMEM`. The former seems to be accurate for Linux, while the latter for macOS (and probably BSDs in general). But to get another pair of eyes on this, @sunfishcode could you offer some advice here please?